src/sidebar/util/annotation-metadata.js

```diff
@@ -149,6 +149,43 @@ function isWaitingToAnchor(annotation) {
   );
 }
 
+/**
+ * Is this annotation a highlight?
+ *
+ * Highlights are generally identifiable by having no text content AND no tags,
+ * but there is some nuance.
+ *
+ * @param {Object} annotation
+ * @return {boolean}
+ */
+function isHighlight(annotation) {
+  // `$highlight` is an ephemeral attribute set by the `annotator` on new
+  // annotation objects (created by clicking the "highlight" button).
+  // It is not persisted and cannot be relied upon, but if it IS present,
+  // this is definitely a highlight (one which is not yet saved).
+  if (annotation.$highlight) {
+    return true;
+  }
+
+  if (isNew(annotation)) {
```

Comment:
Like some other "booleans" in the client app logic, whether or not an annotation is a highlight (or, e.g. an orphan) is implicitly treated as tri-state (true / false / undefined)...but I've made it binary/boolean here even if it's a harsh false.